### PR TITLE
Enable deployments on orgs with encrypted Account.Name and Contact.LastName

### LINKED ISSUE #4465: Encrypted Account Name Causing Errors in CMT_MetadataAPI_TEST
When the Account Name is encrypted via Shield Platform Encryption, the [CMT_MetadataAPI_Test](https://github.com/SalesforceFoundation/Cumulus/blob/master/src/classes/CMT_MetadataAPI_TEST.cls) class throws the following error:

```
Description FROM Account WHERE Name = :SUCCESS_HANDLER_EXECUTED
 ^
ERROR at Row:1:Column:49
field 'Name' can not be filtered in a query call
```

### Steps to Repeat
1. Create Permission Set with the "Manage Encryption Keys" System Permission.
2. Assign the Permission Set to your User record.
3. Go to Setup --> Key Management and click the Generate Tenant Secret button.
4. Go to Setup --> Encryption Policy and click the Encrypt Fields link.
5. Click Edit and check off the Account Name.
6. Save
7. You will receive an email with an error that says: 
```
Encryption enablement has failed for the following fields:
Account.Account Name: 
    Apex Class: CMT_MetadataAPI_TEST, line 70, column 34: field 'Name' can not be filtered in a query call
    Apex Class: CMT_MetadataAPI_TEST, line 78, column 20: field 'Name' can not be filtered in a query call
```

-----

I'll provide resources for how we've resolved this in the past on the generated Work Item.

### PR DESCRIPTION
**Pull request by RadSFDeveloper**
# Critical Changes

# Changes

# Issues Closed
- Fixes #4465 

# New Metadata

# Deleted Metadata

### CODE REVIEW

**Comment by RadSFDeveloper** [comment]
**lurch: attach W-034958

**Comment by LurchTheButler** [comment]
Tracking <a href="https://foundation.my.salesforce.com/a2x1E000001C8dRQAS" target="_blank">W-034958</a>

File: src/classes/CMT_MetadataAPI_TEST.cls [hunk 1]
```diff
@@ -123,10 +123,40 @@ private class CMT_MetadataAPI_TEST {
...
 
-        List<Contact> contacts = [SELECT Id, Lastname FROM Contact WHERE Lastname = :ERROR_HANDLER_EXECUTED];
+        List<Contact> contacts = getContacts(ERROR_HANDLER_EXECUTED);
         System.assertEquals(1, contacts.size(), 'A single contact should be created.'); 
     }
 
+    /*************************************************************************************************************
+     * @description Returns accounts matching the provided name
+     **/
+    private static List<Account> getAccounts(String searchName) {
```
Comment (force2b) [inline src/classes/CMT_MetadataAPI_TEST.cls]: Nice simple resolution here. No need to call to the query/sosl service given that the unit test is working with only a tiny number of test records.

**Comment by RadSFDeveloper** [comment]
> Great job. Have you tried testing in an encrypted org yet?

Yes, I've enabled encryption on the scratch org and encrypted Account.Name and Contact.Name fields per instructions in the issue #4465 . More info on encryption steps in [Trailhead](https://trailhead.salesforce.com/en/content/learn/modules/spe_admins/spe_admins_set_up).

**Comment by RadSFDeveloper** [comment]
@rponti-sforg , the PR is ready for testing.

**Comment by rponti-sforg** [comment]
@RadSFDeveloper @force2b I am not sure what happened here.  There was the sometimes typical branch is 'out of date' message. I clicked the 'Update Branch' button, as I have done in the past.  Now the updated PR is saying I merged a commit into master and I deleted the branch.  I didn't think that 'updating' the branch did that?  Do we have to revert anything?

**Comment by force2b** [comment]
@davisagli Any thoughts on might have happened here?

**Comment by davisagli** [comment]
@rponti-sforg you're saying you didn't intentionally merge this PR?

**Comment by rponti-sforg** [comment]
Correct.  I 'closed' the WI.  Then went to the PR.  Clicked the 'Update Branch' button.  Let it do it's thing.  Was on other tabs doing work.  Came back to this PR tab.  PR was merged and branch deleted.  The one thing I do know for sure, I did not click 'Merge' or 'Delete Branch' button.  I physically did not do that.

**Comment by davisagli** [comment]
Hmm, then that shouldn't have happened. I can do some experiments in another repo and see if this looks like a github bug.

In the meantime, is it okay that this was merged? Or do you need help resetting master to its state from before the merge?

**Comment by rponti-sforg** [comment]
> Hmm, then that shouldn't have happened. I can do some experiments in another repo and see if this looks like a github bug.
> 
> In the meantime, is it okay that this was merged? Or do you need help resetting master to its state from before the merge?

Thank you @davisagli  It is very odd.  Some legit cursor / click magic had to have happened on my end for merge and delete have taken place.  

It is okay to merge.  I am fine with that; testing is complete in scratch org.  I just wanted to make sure with @force2b @RadSFDeveloper that is was indeed okay to merge given that fact we also want to 'beta' test this with encryption turned on when the beta is deployed / installed.  There's another discussion on that, with @davidmreed assisting.  

The initial issue was that customers were not able to install an org with encryption turned on due to an Apex Class Test class error.  I *think* we are good (I know testing is complete and passed) ... just wanted to be sure.

**Comment by davisagli** [comment]
@rponti-sforg in order to test this in a beta it needs to be already merged, so let's at least leave it as is until that testing is done.

**Comment by force2b** [comment]
Thank you both. If possible, a great test would be to spin up a scratch org that has the current production version installed, enable Platform Encryption, and then upgrade to the beta. Though, I think the bug in the 3.159 version that this fixes might actually prevent us from enabling Platform Encryption with 3.160 installed. 

Is it possible to set up a scratch org with NPSP 3.158 installed, then enable Platform Encryption with the Account.Name field encrypted, and then install the 3.161 beta with this fix?

**Comment by davidmreed** [comment]
@force2b I can help set up that org.

**Comment by davidmreed** [comment]
@force2b @RadSFDeveloper I confirmed that all tests in `CMT_MetadataAPI_TEST` pass in a Platform Encryption Beta Org both with and without the Name fields encrypted using NPSP 3.161 (Beta 7).

**Pull request merged**